numpy/_core/src/umath/string_buffer.h

```diff
@@ -1593,4 +1593,43 @@ string_zfill(Buffer<enc> buf, npy_int64 width, Buffer<enc> out)
 }
 
 
+template <ENCODING enc>
+static inline void
+string_partition(Buffer<enc> buf1, Buffer<enc> buf2, npy_int64 idx,
+                 Buffer<enc> out1, Buffer<enc> out2, Buffer<enc> out3,
+                 npy_intp *final_len1, npy_intp *final_len2, npy_intp *final_len3,
+                 STARTPOSITION pos)
+{
```

Comment:
we could add an `assert(enc != ENCODING::UTF8)` or maybe just a comment for future readers that this doesn't work for UTF-8 data.

Reply:
Don't have a strong opinion on this. asserting would probably just make it immediately clear that this is not used by the stringdtype implementation.